ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/ExceptionLoggingInterceptor.cs

```diff
@@ -0,0 +1,146 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Grpc.Core.Interceptors;
+using OliveHelpsLDK.Logging;
+
+namespace OliveHelpsLDK
+{
+    public class ExceptionLoggingInterceptor : Interceptor
+    {
+        private readonly ILogger _logger;
+
+        internal ExceptionLoggingInterceptor(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public override AsyncUnaryCall<TResponse> AsyncUnaryCall<TRequest, TResponse>(TRequest request,
+            ClientInterceptorContext<TRequest, TResponse> context,
+            AsyncUnaryCallContinuation<TRequest, TResponse> continuation)
+        {
+            var call = base.AsyncUnaryCall(request, context, continuation);
+            
+            var logger = new MethodExceptionLogger<TRequest, TResponse>(context, _logger);
+
+            return WrapCall(call, logger);
+        }
+
+        public override AsyncServerStreamingCall<TResponse> AsyncServerStreamingCall<TRequest, TResponse>(
+            TRequest request,
+            ClientInterceptorContext<TRequest, TResponse> context,
+            AsyncServerStreamingCallContinuation<TRequest, TResponse> continuation)
+        {
+            var call = base.AsyncServerStreamingCall(request, context, continuation);
+            
+            var logger = new MethodExceptionLogger<TRequest, TResponse>(context, _logger);
+            
+            return WrapCall(call, logger);
+        }
+
+        private static AsyncUnaryCall<TResponse> WrapCall<TResponse>(AsyncUnaryCall<TResponse> call, ILogger logger)
+        {
+            return new AsyncUnaryCall<TResponse>(WrapContinueHandler(call.ResponseAsync, logger), call.ResponseHeadersAsync, call.GetStatus, call.GetTrailers, call.Dispose);
+        }
+        
+        private static AsyncServerStreamingCall<TResponse> WrapCall<TResponse>(AsyncServerStreamingCall<TResponse> call, ILogger logger)
+        {
+            return new AsyncServerStreamingCall<TResponse>(WrapStreamHandler(call.ResponseStream, logger), call.ResponseHeadersAsync, call.GetStatus, call.GetTrailers, call.Dispose);
+        }
+
+        private static Task<TResponse> WrapContinueHandler<TResponse>(Task<TResponse> task, ILogger logger)
+        {
+            return task.ContinueWith((action) =>
+            {
+                if (action.Exception == null) return action.Result;
+                action.Exception.Handle((exception) =>
+                {
+                    if (exception is RpcException)
+                    {
+                        logger.Error(exception.Message);
```

Comment:
I'm going to have this actually log EVERY exception and then see if there is a way to propagate them without losing them inside other continuations, like the LoggedParser, etc.